File: zig/difference-of-squares/difference_of_squares.zig
const std = @import("std");

pub fn squareOfSum(number: usize) usize {
    var sum: usize = 0;
    for (1..number + 1) |n| {
        sum += n;
    }
    return std.math.pow(usize, sum, 2);
}

pub fn sumOfSquares(number: usize) usize {
    var sum: usize = 0;
    for (1..number + 1) |n| {
        sum += std.math.pow(usize, n, 2);
    }
    return sum;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}

File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(number):
    sum = 0
    for n in range(number+1):
        sum += n

    return pow(sum, 2)


def sum_of_squares(number):
    sum = 0
    for n in range(number+1):
        sum += pow(n, 2)

    return sum


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)
